src/algopy_testing/models/contract.py

```diff
@@ -46,14 +72,46 @@ def _get_local_states(self) -> dict[str, algopy.LocalState[Any]]:
 
         return local_states
 
+    def _get_global_states(self) -> dict[str, algopy.GlobalState[Any]]:
+        import algopy
+
+        global_states = {
+            name: value
+            for name, value in vars(self).items()
+            if isinstance(value, algopy.GlobalState)
+        }
+
+        return global_states
+
+    def _count_state_types(
+        self, states: dict[str, algopy.GlobalState[Any]] | dict[str, algopy.LocalState[Any]]
+    ) -> tuple[int, int]:
+        import algopy
+
+        uint_count = sum(1 for state in states.values() if is_instance(state.type_, algopy.UInt64))
```

Comment:
will need to check for more than just `algopy.UInt64` see other comment for some examples

Reply:
added uintn and biguintn, lmk if there are any other types worth adding

Reply:
added asset and application as well given that those represent int ids